modules/nomad-cluster/main.tf

```diff
@@ -165,3 +163,21 @@ data "aws_iam_policy_document" "instance_role" {
     }
   }
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE ASG ATTACHMENT FOR EVERY PRE-DEFINED LOAD BALANCER OR TARGET GROUP
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_autoscaling_attachment" "elb" {
+  count                  = "${length(var.load_balancers)}"
+
+  autoscaling_group_name = "${aws_autoscaling_group.autoscaling_group.name}"
+  elb                    = "${var.load_balancers[count.index]}"
+}
+
+resource "aws_autoscaling_attachment" "target_group" {
+  count                  = "${length(var.target_group_arns)}"
```

Comment:
Crap, I just realized that the load balancer names and target group ARNs are likely to be output attributes from a resource. That means we cannot use them in `count` due to a Terraform limitation (see https://github.com/hashicorp/terraform/issues/4149). 

So we have two options:

1. Don't create `aws_autoscaling_attachment` in this module at all. This is backwards incompatible and more inconvenient for users.

1. Add `num_target_group_arns` and `num_load_balancers` parameters that must be hard-coded to the length of `var.target_group_arns` and `var.load_balancers` by the user. This muddles the API with a param that's not DRY and hard to maintain.

Sigh. I'm guessing option #1 is better. As you had in your PR originally, we should remove `var.load_balancers` and `var.target_group_arns`, document the user should use `aws_autoscaling_attachment` to attach load balancers, and perhaps add that to the example code.

Reply:
Sure. I'll revert the changes. We can keep the examples I've provided in the opening of the PR.

I'm not sure what examples you are thinking off -- the current examples don't deploy or use any ASG.

Reply:
At the very least, we should update the docs to indicate how to use a load balancer with this code by using the `aws_autoscaling_attachment` resource. Adding an example of doing the same would be nice too, but is not required.